src/lib/device-online-state/index.ts

```diff
@@ -0,0 +1,282 @@
+import * as Bluebird from 'bluebird';
```

Comment:
I don't see any other files in this folder except this `index.ts` one, it's better to move it into the parent directory.